cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/NewTwoWindingsTransformerConversion.java

```diff
@@ -0,0 +1,426 @@
+        TapChangerConversion ratioTapChanger2 = null;
+        TapChangerConversion phaseTapChanger2 = null;
+
+        if (alternative.isXfmr2RatioPhaseEnd1()) {
+            ratioTapChanger1 = combineTapChangers(cgmesT2xModel.end1.ratioTapChanger, cgmesT2xModel.end2.ratioTapChanger);
+            phaseTapChanger1 = combineTapChangers(cgmesT2xModel.end1.phaseTapChanger, cgmesT2xModel.end2.phaseTapChanger);
+        } else if (alternative.isXfmr2RatioPhaseEnd2()) {
+            ratioTapChanger2 = combineTapChangers(cgmesT2xModel.end2.ratioTapChanger, cgmesT2xModel.end1.ratioTapChanger);
+            phaseTapChanger2 = combineTapChangers(cgmesT2xModel.end2.phaseTapChanger, cgmesT2xModel.end1.phaseTapChanger);
+        } else if (alternative.isXfmr2RatioPhaseEnd1End2()) {
+            ratioTapChanger1 = cgmesT2xModel.end1.ratioTapChanger;
+            phaseTapChanger1 = cgmesT2xModel.end1.phaseTapChanger;
+            ratioTapChanger2 = cgmesT2xModel.end2.ratioTapChanger;
+            phaseTapChanger2 = cgmesT2xModel.end2.phaseTapChanger;
+        } else {
+            if (cgmesT2xModel.end1.xIsZero) {
+                ratioTapChanger1 = combineTapChangers(cgmesT2xModel.end1.ratioTapChanger, cgmesT2xModel.end2.ratioTapChanger);
+                phaseTapChanger1 = combineTapChangers(cgmesT2xModel.end1.phaseTapChanger, cgmesT2xModel.end2.phaseTapChanger);
+            } else {
+                ratioTapChanger2 = combineTapChangers(cgmesT2xModel.end2.ratioTapChanger, cgmesT2xModel.end1.ratioTapChanger);
+                phaseTapChanger2 = combineTapChangers(cgmesT2xModel.end2.phaseTapChanger, cgmesT2xModel.end1.phaseTapChanger);
+            }
+        }
+
+        if (alternative.isXfmr2PhaseNegate()) {
+            negatePhaseTapChanger(phaseTapChanger1);
+            negatePhaseTapChanger(phaseTapChanger2);
+        }
+
+        AllTapChanger allTapChanger = new AllTapChanger();
+        allTapChanger.ratioTapChanger1 = ratioTapChanger1;
+        allTapChanger.phaseTapChanger1 = phaseTapChanger1;
+        allTapChanger.ratioTapChanger2 = ratioTapChanger2;
+        allTapChanger.phaseTapChanger2 = phaseTapChanger2;
+
+        return allTapChanger;
+    }
+
+    private AllShunt shuntAlternative(CgmesT2xModel cgmesT2xModel, Conversion.Config alternative) {
+        double g1 = 0.0;
+        double b1 = 0.0;
+        double g2 = 0.0;
+        double b2 = 0.0;
+        if (alternative.isXfmr2ShuntEnd1()) {
+            g1 = cgmesT2xModel.end1.g + cgmesT2xModel.end2.g;
+            b1 = cgmesT2xModel.end1.b + cgmesT2xModel.end2.b;
+        } else if (alternative.isXfmr2ShuntEnd2()) {
+            g2 = cgmesT2xModel.end1.g + cgmesT2xModel.end2.g;
+            b2 = cgmesT2xModel.end1.b + cgmesT2xModel.end2.b;
+        } else if (alternative.isXfmr2ShuntEnd1End2()) {
+            g1 = cgmesT2xModel.end1.g;
+            b1 = cgmesT2xModel.end1.b;
+            g2 = cgmesT2xModel.end2.g;
+            b2 = cgmesT2xModel.end2.b;
+        } else {
```

Comment:
What is this case?

Reply:
This case is now auto-documented after moving interpretation options to enum lists